macros/src/console_fmt.rs

```diff
@@ -49,20 +56,31 @@ impl ConsoleFmt for bool {
             FormatSpec::String => self.pretty(),
             FormatSpec::Object => format!("'{}'", self.pretty()),
             FormatSpec::Number => (*self as i32).to_string(),
-            FormatSpec::Integer => String::from("NaN"),
+            FormatSpec::Integer | 
+                FormatSpec::Exponential | FormatSpec::Hexadecimal => String::from("NaN"),
         }
     }
 }
 
 impl ConsoleFmt for U256 {
-    fn fmt(&self, _spec: FormatSpec) -> String {
-        self.pretty()
+    fn fmt(&self, spec: FormatSpec) -> String {
+        match spec {
+            FormatSpec::String | FormatSpec::Object
+                | FormatSpec::Number | FormatSpec::Integer => self.pretty(),
+            FormatSpec::Hexadecimal => format!("0x{:x}", *self),
+            FormatSpec::Exponential => format!("0x{:x}", *self)
```

Comment:
if they behave identically, why even have two additional placeholders?

Reply:
I see

there's format_units, which is perhaps helpful

https://github.com/gakonst/ethers-rs/blob/72428503cfb0855579d47f691b48b2d65ad8195b/ethers-core/src/utils/mod.rs#L168